plugin/trino-loki/src/main/java/io/trino/loki/LokiRecordSet.java

```diff
@@ -43,15 +44,22 @@ public LokiRecordSet(LokiClient lokiClient, LokiSplit split, List<LokiColumnHand
         }
         this.columnTypes = types.build();
 
-        // TODO: get query and start from table
-        final String query = "{source=\"stderr\"}";
-        final Long start = 1722852098L - (15 * 60 * 1000); // Start and end should be part of split
-        final Long end = 1722852098L;
+        Long end = now();
+        Long start = end - ONE_HOUR;
+        //if split.end().getEpochMillis() != 0 {}
```

Comment:
This might be removed

Reply:
We actually should use `end` if it's available. I'll fix that